proposed/0023-crust-mock/0023-crust-mock.md

```diff
@@ -0,0 +1,106 @@
+- Feature Name: Crust mock
+- Type: feature
+- Related components: crust, routing
+- Start Date: 18-02-2016
+- RFC PR:
+- Issue number:
+
+# Summary
+
+Drop-in mock replacement for Crust to make it easier to unit test libraries and
+applications that use Crust.
+
+# Motivation
+
+Testing Crust-depending libraries on real network (even on LAN or localhost) is
+cumbersome, slow and potentially expensive. A drop in mock replacement for Crust
+could make these test simpler and faster, make it easier to diagnose failures,
+and help cover edge cases difficult to set up on real network.
+
+# Detailed design
+
+This section details one possible implementation of the Crust Mock.
+
+## Changes to crust
+
+A new trait `ConnectionService` MUST be defined. It MUST have all the relevant methods of `crust::Service`:
+
+```rust
+pub trait ConnectionService {
+  fn start_listening_tcp();
+  fn start_listening_utp();
+  fn start_service_discovery();
+  fn prepare_connection_info();
+  fn connect();
+  fn disconnect();
+  fn send();
```

Comment:
OK. I'll update the signatures of these functions to be the same as in `crust::Service`.